File: app/ios/KeyboardExtension/AI/AiProviderFactory.swift
import Foundation

class AiProviderFactory {
    static func createProvider(for providerType: String) -> AiProvider? {
        switch providerType.lowercased() {
        case "openai": return OpenAiProvider()
        case "gemini": return GeminiProvider()
        case "groq": return GroqProvider()
        case "openrouter": return OpenRouterProvider()
        default: return nil
        }
    }
}

File: app/android/app/src/main/kotlin/com/pk/atfix/ai/AiProviderFactory.kt
package com.pk.atfix.ai

object AiProviderFactory {
    fun getProvider(providerName: String): AiProvider {
        return when (providerName.lowercase()) {
            "openai" -> OpenAiProvider()
            "gemini" -> GeminiProvider()
            "openrouter" -> OpenRouterProvider()
            "groq" -> GroqProvider()
            else -> OpenAiProvider()
        }
    }
}
